src/main/java/org/zendesk/client/v2/Zendesk.java

```diff
@@ -3783,6 +3965,27 @@ private TemplateUri tmpl(String template) {
     return new TemplateUri(url + template);
   }
 
+  private Uri cbp(String path, Param... queryParameters) {
+    return cbp(path, Arrays.asList(queryParameters));
+  }
+
+  private Uri cbp(String path, Iterable<Param> queryParameters) {
+    Objects.requireNonNull(path, "Path cannot be null");
+    if (path.indexOf('?') != -1) {
+      throw new IllegalArgumentException("Path cannot contain a query string");
+    }
+    // No need to encode parameters. Handled by AHC.
+    // See
+    // https://github.com/AsyncHttpClient/async-http-client/blob/async-http-client-project-2.12.3/client/src/main/java/org/asynchttpclient/RequestBuilderBase.java#L598-L609
+    StringBuilder sb =
+        new StringBuilder(url).append(path).append("?page[size]=").append(cbpPageSize);
```

Comment:
This project relies on https://github.com/damnhandy/Handy-URI-Templates to build URIs. It removes the hassle of manually building the string (which is error prone too). See the README in the link I provided, it contains useful examples.

Reply:
The reason why the uri templates were not used is that they do not support `page[size]` as a parameter name:
```
  UriTemplate invalid = UriTemplate.fromTemplate("https://yourdomain.zendesk.com/api/v2/groups?{page[size],foo}");
  invalid.set("page[size]", 2);
  invalid.set("foo", "bar");
  System.out.println(invalid.expand());

```
produces `com.damnhandy.uri.template.MalformedUriTemplateException: The variable name page[size] contains invalid characters`

I believe this is the same reason this [code exists](https://github.com/cloudbees-oss/zendesk-java-client/blob/master/src/main/java/org/zendesk/client/v2/Zendesk.java#L1074-L1076).

It may not be an issue though once I get rid of the ability to pass query parameters.